src/pretix/base/datasync/datasync.py

```diff
@@ -0,0 +1,425 @@
+    """
+    def next_retry_date(self, sq):
+        return now() + timedelta(hours=1)
+
+    """
+    Optionally override this method to exclude certain orders from sync by returning False
+    """
+    def order_valid_for_sync(self, order):
+        return True
+
+    """
+    Implementations must override this property to provide the data mappings as a list of objects.
+
+    They can return instances of the StaticMapping namedtuple defined above, or create their own
+    class (e.g. a Django model).
+
+    The returned objects must have at least the following properties:
+    - pk:                   unique identifier
+    - pretix_model:         which pretix model to use as data source in this mapping. possible values are
+                            the keys of sourcefields.AVAILABLE_MODELS
+    - external_object_type: destination object type in the target system. opaque string of maximum 128 characters.
+    - pretix_pk:            which pretix data field should be used to identify the mapped object. any
+                            DataFieldInfo.key returned by sourcefields.get_data_fields for the combination of
+                            Event and pretix_model
+    - external_pk:          destination identifier field in the target system
+    - property_mapping:     mapping configuration as generated by PropertyMappingFormSet.to_property_mapping_json()
+    """
+    @property
+    def mappings(self):
+        raise NotImplementedError
+
+    def sync_queued_orders(self, queued_orders):
+        for sq in queued_orders:
+            try:
+                mapped_objects = self.sync_order(sq.order)
+            except SyncConfigError as e:
+                logger.warning(
+                    f"Could not sync order {sq.order.code} to {type(self).__name__} (config error)",
+                    exc_info=True,
+                )
+                sq.order.log_action("pretix.event.order.data_sync.failed", {
+                    "provider": self.identifier,
+                    "error": e.messages,
+                    "full_message": e.full_message,
+                })
+                sq.delete()
+            except Exception as e:
+                # TODO: different handling per Exception, or even per HTTP response code?
+                #       otherwise, SyncProviders should always throw SyncConfigError in non-recoverable situations
+                sq.failed_attempts += 1
+                sq.not_before = self.next_retry_date(sq)
+                logger.exception(
+                    f"Could not sync order {sq.order.code} to {type(self).__name__} (transient error, attempt #{sq.failed_attempts})"
+                )
+                if sq.failed_attempts >= self.max_attempts:
+                    sentry_sdk.capture_exception(e)
+                    sq.order.log_action("pretix.event.order.data_sync.failed", {
+                        "provider": self.identifier,
+                        "error": [_("Maximum number of retries exceeded.")],
```

Comment:
This code runs in the cronjob, so it always runs with English locale, therefore the translation tags are probably pointless